### PR TITLE
Surface errors during ccz creation

### PR DESCRIPTION
**Pull request by orangejenny**
`iter_media_files` and `iter_app_files` both capture errors, but so far as I can tell, those errors just get thrown away.

@mkangia / @snopoke 
code buddy @dannyroberts 

screen shot 2019-02-15 at 9 36 57 pm

### CODE REVIEW

**Comment by orangejenny** [comment]
This is in the service of https://dimagi-dev.atlassian.net/browse/ICDS-291 - I'm hoping there are already errors being thrown when multimedia is missing and we've just been ignoring them.

File: corehq/apps/hqmedia/tasks.py [hunk 1]
```diff
@@ -152,6 +154,9 @@ def build_application_zip(include_multimedia_files, include_index_files, app,
                     z.writestr(path, data, file_compression)
                     progress += file_progress / file_count
                     DownloadBase.set_progress(build_application_zip, progress, 100)
+        if errors:
+            build_application_zip.update_state(state=states.FAILURE, meta={'errors': errors})
+            raise Ignore()  # We want the task to fail hard, so ignore any future updates to it
```
Comment (snopoke) [inline corehq/apps/hqmedia/tasks.py]: :+1:

**Review by snopoke — APPROVED** [review]

**Comment by mkangia** [comment]
This is great.
Is there a way to make those error messages more user friendly?
@orangejenny

**Comment by orangejenny** [comment]
> Is there a way to make those error messages more user friendly?

Not at this point. `iter_index_files` does an `except Exception` so that could be anything, and I don't know what causes the `NameError` raised in `iter_media_files`.

**Pull request merged**